client/flutter/lib/main.dart

```diff
@@ -73,6 +74,25 @@ class _MyAppState extends State<MyApp> {
       },
     );
 
+    updateFCMToken();
+  }
+
+  // Ask firebase for a token on every launch.
+  // If the token is different from what we have stored...update it.
+  updateFCMToken() async {
+    // Only check if notifications are enabled. 
+    bool notificationsEnabled =
+        await UserPreferences().getNotificationsEnabled();
+
+    if (notificationsEnabled) {
+      await _firebaseMessaging.getToken().then((String token) async {
```

Comment:
Need to set storage only after telling server.  Otherwise on failed server store we'll assume incorrectly we dont need to update server.

```suggestion
      final token = await _firebaseMessaging.getToken();
      String storedToken = await UserPreferences().getFCMToken();
      if (token != storedToken) {
        await WhoService.putDeviceToken(token);
        await UserPreferences().setFCMToken(token);
      }
```